common4j/src/main/com/microsoft/identity/common/java/net/UrlConnectionHttpClient.java

```diff
@@ -270,7 +270,7 @@ private static HttpRequest constructHttpRequest(@NonNull HttpClient.HttpMethod h
                                                     byte[] requestContent) {
 
         // Apply special backcompat behaviors for PATCH, if reqd
-        if (HttpClient.HttpMethod.PATCH == httpMethod) {
+        if (HttpClient.HttpMethod.PATCH == httpMethod && !requestUrl.getPath().contains("upgradeRegistration")) {
```

Comment:
Can we create overload if method which accepts let caller pass if PATCH -> POST with appropriate header added? UpgradeDeviceRegistrationRequest can call it.

Reply:
I override the patch method (which is a separate method) to keep the implementation to not convert it to Post and use PATCH.